tests/unit/test_cli_controller.py

```diff
@@ -144,3 +151,51 @@ def test_delete_many_no_id():
     result = runner.invoke(app, ["delete", "many"])
     assert result.exit_code == 3
     assert "Please provide at least 2 task id's to delete." in result.stdout
+
+
+# Sample data to be returned by the mock read_tasks_file function
+sample_tasks = [
+    {
+        "UUID": "123",
+        "name": "Task One",
+        "date": "2023",
+        "priority": "Low"
+    }
+]
+
+
+@pytest.fixture
+def mock_read_tasks_file():
+    """
+    This will mock the read_tasks_file function.
+    """
+    with patch("todo_cli.cli.cli_controller.read_tasks_file") as mock:
+        mock.return_value = sample_tasks
+        yield mock
+
+
+@pytest.mark.parametrize("format_type,expected_output", [
+    (Formats.JSON,
+     """[
+    {
+        "UUID": "123",
+        "name": "Task One",
+        "date": "2023",
+        "priority": "Low"
+    }
+]"""
+     ),
+    (Formats.CSV, 'UUID,name,date,priority\r\n123,Task One,2023,Low\r\n')
+])
+def test_export(mock_read_tasks_file, format_type, expected_output):  # pylint: disable=redefined-outer-name
+    """
+    This will test the export function.
+    """
+    assert mock_read_tasks_file is not None
+    mock_file = io.StringIO()
+
+    export(file_format=format_type, file=mock_file)
+
+    mock_file.seek(0)
+    output = mock_file.read()
+    assert output == expected_output
```

Comment:
Follow up to my other comment, this is very different way of running the test. We usually use the runner.invoke so a wondering why we deviated from that which is fine but then we should adjust the others to try and be consistent with this, if this way is preferred.